Src/main.c

```diff
@@ -106,15 +112,25 @@ int main(void)
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
+  const uint8_t TIMER_LOW = 1;
+  const uint8_t TIMER_HIGH = 2;
+
   while (1)
   {
-  /* USER CODE END WHILE */
 
+  /* USER CODE END WHILE */
+    
   /* USER CODE BEGIN 3 */
+    uint16_t adcValue = HAL_ADC_GetState(&hadc);
+    float advValPercentage = adcValue/4096.0; // Converting the converted digital value into a percentage
```

Comment:
max adc value is actually 4095, because there are 4096 possible values and one of them is 0. Don't mean to give away the answer, but you were very close lol